cmd/brew-test-bot.rb

```diff
@@ -977,7 +980,10 @@ def test_ci_upload(tap)
 
     if (pr = ENV["UPSTREAM_PULL_REQUEST"])
       pull_pr = "https://github.com/#{tap.user}/homebrew-#{tap.repo}/pull/#{pr}"
-      safe_system "brew", "pull", "--clean", pull_pr
+      args = []
+      args << "--tap=#{@tap}" unless @tap.nil?
+      args << pull_pr
+      test "brew", "pull", "--clean", *args
```

Comment:
Can you explain why both of these are needed, it's not clear? It seems like this should be autodetected by `brew pull` and probably fixed there. Additionally, this is a copy-paste so would be good to avoid duplication here.